include/rmm/mr/device/cuda_memory_resource.hpp

```diff
@@ -65,6 +66,7 @@ class cuda_memory_resource final : public device_memory_resource {
    */
   void* do_allocate(std::size_t bytes, cuda_stream_view) override
   {
+    RMM_FUNC_RANGE();
```

Comment:
Instead of putting these in every resource, I think it should just go in the `device_memory_resource` base class.

Reply:
Yeah, I'm planning that.  But I'm not going to work on this PR until NVTX is actually released...